src/components/signup/SignUp.js

```diff
@@ -0,0 +1,161 @@
+import "./SignUp.css";
+import { FaTwitter } from "react-icons/fa";
+import { useState } from "react";
+import { axiosInstanceReg } from "../api/axios";
+import Login from "../login/Login";
+
+const SignUp = () => {
+  const [user, setUser] = useState({
+    first_name: "",
+    last_name: "",
+    username: "",
+    email: "",
+    password: "",
+    password_confirmation: "",
+  });
+  const [errMsg, setErrMsg] = useState("");
+  const [success, setSuccess] = useState(false);
+
+  const handleChange = (e) => {
+    setUser((state) => ({
+      ...state,
+      [e.target.name]: e.target.value.trim(),
+    }));
+  };
+
+  const handleSubmit = () => {
+    if (user.password !== user.password_confirmation) {
+      setErrMsg("Password does not match");
+      setSuccess(false);
+    } else {
+      postRequest();
```

Comment:
You using this before it is initialized?

Reply:
will rearrange this